internal/discord/command-handlers.go

```diff
@@ -0,0 +1,314 @@
+func (d *Discord) AddCommands(s *discordgo.Session, event *discordgo.Ready) {
+	fmt.Printf("Initializing Discord...\n")
+
+	for _, discordGuild := range event.Guilds {
+
+		// Adding commands to a list to prepare in bulk
+		var commands []*discordgo.ApplicationCommand
+		commands = append(commands,
+			KickCommand,
+			MuteCommand,
+			UnmuteCommand,
+			BanCommand,
+			UnbanCommand,
+			RemoveNicknameCommand,
+			SetNicknameCommand,
+			SlowmodeCommand,
+			SlowmodeOffCommand,
+			PurgeCommand,
+			ExileCommand,
+			UnexileCommand,
+		)
+
+		fmt.Printf("Adding commands...\n")
+		commandList, err := s.ApplicationCommandBulkOverwrite(event.User.ID, discordGuild.ID, commands)
+		fmt.Printf("List of successfully created commands:\n")
+		for _, command := range commandList {
+			fmt.Printf("\t%v\n", command.Name)
+		}
+		if err != nil {
+			fmt.Printf("Could not add some commands: %v \n", err)
+		}
+	}
+}
+
+var KickCommand = &discordgo.ApplicationCommand{
+	Name:                     "kick",
+	DefaultMemberPermissions: &kickPermission,
+	Description:              "Kick the specified user and notify the user why via DMs.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionUser,
+			Name:        "user",
+			Description: "User being kicked",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "reason",
+			Description: "Reason for kick",
+			Required:    true,
+		},
+	},
+}
+
+var MuteCommand = &discordgo.ApplicationCommand{
+	Name:                     "mute",
+	DefaultMemberPermissions: &mutePermission,
+	Description:              "Mute the specified user.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionUser,
+			Name:        "user",
+			Description: "User being muted",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "duration",
+			Description: "Duration of mute (e.g \"1m\", \"1h\", \"1d\")",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "reason",
+			Description: "Reason for mute",
+			Required:    true,
+		},
+	},
+}
+
+var UnmuteCommand = &discordgo.ApplicationCommand{
+	Name:                     "unmute",
+	DefaultMemberPermissions: &mutePermission,
+	Description:              "Unmute the specified user.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionUser,
+			Name:        "user",
+			Description: "User being unmuted",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "reason",
+			Description: "Reason for unmute",
+			Required:    true,
+		},
+	},
+}
+
+var BanCommand = &discordgo.ApplicationCommand{
+	Name:                     "ban",
+	DefaultMemberPermissions: &banPermission,
+	Description:              "Ban the specified user.",
```

Comment:
let's make this similar to `KickCommand` and append ` and notify the user why via DMs.`